packages/next-auth/src/providers/email.ts

```diff
@@ -16,8 +15,14 @@ export interface SendVerificationRequestParams {
 
 export interface EmailConfig extends CommonProviderOptions {
   type: "email"
-  // TODO: Make use of https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html
-  server: string | SMTPTransportOptions
+  server: `smtp://${string}:${string}@${string}.${string}:${number}` | {
```

Comment:
This seems stricter than before. For the object, I think it's best to keep using the TS type from `nodemailer`. There are other properties there that could be useful for the developer 🤔 https://nodemailer.com/smtp/

For the string: 
- We could simplify the `host` bit to `string` instead of `${string}.${string}`. That's because sometimes the host can have more than one dot (subdomains etc...).
- Can we have optional query params at the end?
- Can we add support for `smtps` too? 
Let me know what you think. 🙌

Reply:
great feedback! unfortunately at the moment the types from `nodemailer` don't seem to work - i will check again though

re: `${string}.${string}` - this should already accept multiple dots ([link to TS Playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAysBOBLAdgcwCIHthyWqAvFAAYAkA3gM4IqoC+AdBdXvcQLABQXAxpstSgBDAFywaaLDgmpCUAOQAbTPAgBbBojCUArmvm9+ggEZjctKefxF5AE0zL4DSomAMhaiMAOcA9L6gAAWBKAFoIAA9IHmBw+HgVQwFgKB4zGUsZOXlkbFCUHkUdW1pQllofIA))

definitely good point re: query params at the end and `smtps` - i will add these